### PR TITLE
Interface scream with MCT, Vol. I

### PR DESCRIPTION
**Pull request by bartgol**
This PR sets some things up that in the direction of interfacing with the component coupler.

### CODE REVIEW

**Comment by bartgol** [comment]
@AaronDonahue This is ready for review. It is _not_ the whole deal, but it sets up a few things, including some stuff for cime.

File: components/scream/src/interface/scream_cxx_f90_interface.cpp [hunk 1]
```diff
@@ -1,102 +1,104 @@
...
-  // Set the dummy grid in the UserProvidedGridManager
-  // Recall that this class stores *static* members, so whatever
-  // we set here, will be reflected in the GM built by the factory.
-  auto& upgm = c.create<UserProvidedGridsManager>(); // upgm;
-  upgm.set_grid(std::make_shared<DummyPhysicsGrid>(num_cols));
-  upgm.set_reference_grid("Physics");
-
-  // Create a comm
-  Comm atm_comm (MPI_COMM_WORLD);
+  printf("[scream] reading parameterr from yaml file: %s\n",input_yaml_file);
```
Comment (AaronDonahue) [inline components/scream/src/interface/scream_cxx_f90_interface.cpp:52]: Not major but, "parameter" is misspelled.

**Comment by bartgol** [comment]
> Should the cmake and CIME based changes be reviewed by Jim? Or have you already consulted with him offline?

I haven't been able to talk to Jim yet. He might have taken a few days off. As soon as he gets back, I'll ask him. I don't want to merge without his approval.

**Comment by jgfouca** [comment]
I reviewed the CIME and cmake changes and they look good.

**Comment by bartgol** [comment]
@AaronDonahue I'm working on a somewhat more organized fix for the initialization.

**Comment by bartgol** [comment]
@jgfouca I changed how we handle build type and cmake's default flags for that build. Please, double check if things seem sound to you.

**Comment by AaronDonahue** [comment]
@bartgol , you may also want to change the description of the PR to remove the reference to WIP

**Pull request merged**